pkgs/development/compilers/rust/rustc.nix

```diff
@@ -199,15 +190,23 @@ stdenv.mkDerivation (finalAttrs: {
     ++ optionals (!withBundledLLVM) [
       "--enable-llvm-link-shared"
       "${setBuild}.llvm-config=${llvmSharedForBuild.dev}/bin/llvm-config"
-      "${setHost}.llvm-config=${llvmSharedForHost.dev}/bin/llvm-config"
-      "${setTarget}.llvm-config=${llvmSharedForTarget.dev}/bin/llvm-config"
     ]
+    ++ (
+      if (llvmShared.stdenv.hostPlatform == llvmShared.stdenv.buildPlatform) then
+        [
+          "${setHost}.llvm-config=${llvmShared.dev}/bin/llvm-config"
+        ]
+      else
+        [
+          "${setHost}.llvm-config=${llvmShared.dev}/bin/llvm-config-native"
```

Comment:
I wonder if we should always create llvm-config-native, even in non-cross builds… This is fine for now though!